Add runbook and update codeowners (#26)
Financial-Times/smartlogic-concordance-transformer: runbooks/runbook.md

# UPP - Smartlogic Concordance Transformer

> This service will listen to Kafka for a notification of a change made in Smartlogic, verify whether the change concerns concordance, convert the JSON-LD in the message to a normalised UPP view of a concordance and finally send the JSON to the concordance-rw-dynamodb.

Reviewer: Let's simplify this and keep the long description in the Architecture field.
```suggestion
This service listens for changes made in Smartlogic and persists the resulting concordance changes in the Cloud Neo4j using "concordances-rw-neo4j".
```

## Code

smartlogic-concordance-transform

## Primary URL

> <https://github.com/Financial-Times/smartlogic-concordance-transformer>

Reviewer 2: Shouldn't this be a URL to the service itself not to the github repo?

Author: @ekaneva As far as I know only the public services have such. I'm not sure.  

I don't know the URL, can someone point me it is the right direction? What is the correct value or Where/How to find it.

Reviewer 2: When there is no public URL, the one that is used to hit the service, e.g:
<https://upp-prod-delivery-glb.upp.ft.com/__smartlogic-concordance-transformer/>

Not sure if this is the correct one in this case. Again, someone from @Financial-Times/metadata-team would be more useful here.

Reviewer: It's the publishing cluster GLB URL, otherwise you are correct:
```
https://upp-prod-publish-glb.upp.ft.com/__smartlogic-concordance-transformer/
```

## Service Tier

Bronze

## Lifecycle Stage

Production

## Delivered By

content

## Supported By

content

## Known About By

- elitsa.pavlova
- kalin.arsov
- ivan.nikolov
- marina.chompalova
- miroslav.gatsanoga

## Host Platform

AWS

## Architecture

> This service will listen to Kafka for a notification of a change made in Smartlogic, verify whether the change concerns concordance, convert the JSON-LD in the message to a normalised UPP view of a concordance and finally send the JSON to the concordance-rw-dynamodb.

Reviewer: ```suggestion
This service will listen to the "SmartlogicConcept" Kafka topic for a notification of a change made in Smartlogic, verify whether the change concerns concordance, convert the JSON-LD in the message to a normalised UPP view of a concordance and finally send the JSON to the "concordances-rw-neo4j" service.
```

Checkout the application project repository for further details:
<https://github.com/Financial-Times/smartlogic-concordance-transformer>

## Contains Personal Data

No

## Contains Sensitive Data

No

> 

Reviewer: ```suggestion

## Dependencies

- concordances-rw-neo4j
- upp-kafka
- upp-zookeeper

```

## Failover Architecture Type

> ActiveActive

Reviewer: The service is located in the publishing clusters, so:
```suggestion
ActivePassive
```

## Failover Process Type

FullyAutomated

## Failback Process Type

> FullyAutomated

Reviewer: The service is located in the publishing clusters, so:
```suggestion
Manual
```

## Failover Details

The service is deployed in the Publish clusters. The failover guide for the cluster is located here: <https://github.com/Financial-Times/upp-docs/tree/master/failover-guides/publishing-cluster>

## Data Recovery Process Type

NotApplicable

## Data Recovery Details

The service does not store data, so it does not require any data recovery steps.

## Release Process Type

PartiallyAutomated

## Rollback Process Type

Manual

## Release Details

The release is triggered by making a Github release which is then picked up by a Jenkins multibranch pipeline. The Jenkins pipeline should be manually started in order for it to deploy the helm package to the Kubernetes clusters.

## Key Management Process Type

NotApplicable

## Key Management Details

There is no key rotation procedure for this system.

## Monitoring

Look for the pods in the cluster health endpoint and click to see pod health and checks:

> - <https://upp-prod-publish-eu.upp.ft.com/__health/>
> - <https://upp-prod-publish-us.upp.ft.com/__health/>

Reviewer: ```suggestion
- EU cluster: <https://upp-prod-publish-eu.upp.ft.com/__health/__pods-health?service-name=smartlogic-concordance-transformer>
- US cluster: <https://upp-prod-publish-us.upp.ft.com/__health/__pods-health?service-name=smartlogic-concordance-transformer>
```
